components/camel-swagger-java/src/main/java/org/apache/camel/swagger/RestSwaggerSupport.java

```diff
@@ -345,4 +353,36 @@ private static void setupCorsHeaders(RestApiResponseAdapter response, Map<String
         response.setHeader("Access-Control-Max-Age", maxAge);
     }
 
+    private void setupXForwardedHeaders(Swagger swagger, Map<String, Object> headers) {
+
+        String host = (String) headers.get(HEADER_HOST);
+        if(host != null && host.length() > 0) {
+            swagger.setHost(host);
+        }
+
+        String forwardedPrefix = (String) headers.get(HEADER_X_FORWARDED_PREFIX);
+        if (forwardedPrefix != null && forwardedPrefix.length() > 0) {
+            swagger.setBasePath("/" + forwardedPrefix.replace("/", "") + swagger.getBasePath());
+        }
+
+        String forwardedHost = (String) headers.get(HEADER_X_FORWARDED_HOST);
+        if(forwardedHost != null && forwardedHost.length() > 0) {
+            swagger.setHost(forwardedHost);
+        }
+
+        String proto = (String) headers.get(HEADER_X_FORWARDED_PROTO);
+        if(proto != null && proto.length() > 0) {
+            String[] schemes = proto.split(",");
+            List<Scheme> schs = new ArrayList<>();
+            for(String scheme : schemes) {
+                String trimmedScheme = scheme.trim();
+                schs.add(Scheme.forValue(trimmedScheme));
+            }
+            swagger.setSchemes(schs);
+        }
+        else {
+            swagger.setSchemes(null);
```

Comment:
Do we want to set the `scheme` if there's no `X-Forwarded-Proto` header given?

Reply:
In this case the default is not return anything when proto is not given.